### PR TITLE
Split out JSON Date Converters

### PR DESCRIPTION
**Pull request by WillAyd**
This file is pretty large so trying to make more modular. This separates out conversion routines required for getting from the Python space to JSON, but which don't actually serialize the data

### CODE REVIEW

File: pandas/_libs/src/ujson/python/date_conversions.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+npy_datetime NpyDateTimeToEpoch(npy_datetime dt, NPY_DATETIMEUNIT base);
+
+// Converts a Python object representing a Date / Datetime to ISO format
+// up to precision `base` e.g. base="s" yields 2020-01-03T00:00:00Z
+// while base="ns" yields "2020-01-01T00:00:00.000000000Z"
+// len is mutated to save the length of the returned string
+char *PyDateTimeToIso(PyDateTime_Date *obj, NPY_DATETIMEUNIT base, size_t *len);
+
+// Convert a Python Date/Datetime to Unix epoch with resolution base
+npy_datetime PyDateTimeToEpoch(PyDateTime_Date *dt, NPY_DATETIMEUNIT base);
```
Comment (WillAyd) [inline pandas/_libs/src/ujson/python/date_conversions.h]: As a minor detail I had to modify the signature of this which previously accepted a PyObject and perform a `PyDateTime_Check` call. The problem with moving that into a separate file is that you need to call `PyDateTime_IMPORT` to set an object with static duration in this file separate from what is already in `objToJSON.c` and there's not a great place to really set that. 

We didn't explicitly do anything with the check previously and this aligns better with `PyDateTimeToIso` so I think not a big deal

File: pandas/_libs/src/ujson/python/date_conversions.c [hunk 2]
```diff
@@ -0,0 +1,118 @@
+// Conversion routines that are useful for seralization,
```
Comment (jbrockmendel) [inline pandas/_libs/src/ujson/python/date_conversions.c]: seralization -> serialization

**Comment by jbrockmendel** [comment]
seems like a good refactor.

Out of curiosity, what is the big picture gameplan for the json code?

**Comment by WillAyd** [comment]
Just trying to make more readable / scalable and learn more about C

**Comment by jbrockmendel** [comment]
LGTM cc @jreback

**Comment by jreback** [comment]
can you rebase

**Comment by WillAyd** [comment]
rebased and green

**Comment by jreback** [comment]
lgtm. thanks @WillAyd

**Pull request merged**